marquez_airflow/extractors/__init__.py

```diff
@@ -37,43 +37,90 @@ def __eq__(self, other):
                self.connection_url == other.connection_url
 
     def __repr__(self):
-        return f"Source({self.name!r}, {self.type!r}, {self.connection_url!r})"
+        return f"Source({self.name!r},{self.type!r},{self.connection_url!r})"
 
 
-class Dataset:
-    source: Source = None
-    name = None
-    description = None
-    type = None
+class Field:
+    def __init__(self, name: str, type: str,
+                 tags: List[str] = [], description: str = None):
+        self.name = name
+        self.type = type
+        self.tags = tags
+        self.description = description
+
+    @staticmethod
+    def from_column(column: DbColumn):
+        return Field(
+            name=column.name,
+            type=column.type,
+            description=column.description
+        )
+
+    def __eq__(self, other):
+        return self.name == other.name and \
+               self.type == other.type and \
+               self.tags == other.tags and \
+               self.description == other.description
 
-    def __init__(self, source, name, type, description=None):
+    def __repr__(self):
+        return f"Field({self.name!r},{self.type!r}, \
+                       {self.tags!r},{self.description!r})"
+
+
+class Dataset:
+    def __init__(self, source: Source, name: str, type: DatasetType,
+                 fields: List[Field] = [], description=None):
         self.source = source
         self.name = name
         self.type = type
+        self.fields = fields
         self.description = description
 
     @staticmethod
-    def from_table(source, table):
+    def from_table_only(source: Source, table_name: str,
+                        schema_name: str = None):
         return Dataset(
             type=DatasetType.DB_TABLE,
-            name=table,
+            name=Dataset._to_name(
+                schema_name=schema_name,
+                table_name=table_name
+            ),
             source=source
         )
 
+    @staticmethod
+    def from_table_schema(source: Source, table_schema: DbTableSchema):
+        return Dataset(
+            type=DatasetType.DB_TABLE,
+            name=Dataset._to_name(
+                schema_name=table_schema.schema_name,
+                table_name=table_schema.table_name
+            ),
+            source=source,
+            fields=[
+                # We want to maintain column order using ordinal position.
+                Field.from_column(column) for column in sorted(
+                    table_schema.columns, key=lambda x: x.ordinal_position
+                )
+            ]
+        )
+
+    @staticmethod
+    def _to_name(table_name: str, schema_name: str = None):
+        # Prefix the table name with the schema name using
+        # the format: {table_schema}.{table_name}.
+        return f"{schema_name}.{table_name}" if schema_name else table_name
```

Comment:
to have canonical names we need to use the default schema of the current connection when the schema is not provided.

Reply:
I left the `schema_name` param here optional (for now) as not all tables will have schemas. For example, `BigQuery` table naming uses: `{projectId}.{datasetId}.{tableId}`. We'll want to define a set of naming conventions for how to handle naming form various source.

Reply:
all tables in postgres will have schemas. when using just the table name in a query, the database is assuming the "current" schema of the connection. both "current schema"."table" and "table will point to the same table. We need to give the corresponding dataset a consistent name independently of whether the user is using the fully qualified name or not.

Reply:
(but this is not related to this PR, so can be addressed independently)

Reply:
Ahh, good point. We may also want to add the name of source registered with Marquez as a prefix to the dataset name as well.